### PR TITLE
Sync ElasticPress to 3.6.5

### PR DESCRIPTION
**Pull request by sjinks**
## Description

Synchronize `develop` with the `3.6.5` tag of the upstream.

## Checklist

Please make sure the items below have been covered before requesting a review:

- [ ] This change works and has been tested locally (or has an appropriate fallback).
- [ ] This change works and has been tested on a Go sandbox.
- [ ] This change has relevant unit tests (if applicable).
- [ ] This change has relevant documentation additions / updates (if applicable).
- [ ] This change has the fix PRed upstream (if applicable). If not applicable, it has the relevant "// VIP: reason for the discrepancy with upstream" comment in places where the code is discrepant.

## Steps to Test

The Ci ideally should pass.

### CODE REVIEW

**Comment by sjinks** [comment]
Conflicts:
* includes/classes/Indexable/User/User.php: conflicts in `@codingStandardsIgnoreXXX` tags; replaced with the upstream implementation (`phpcs:disable`) [[link](https://github.com/Automattic/ElasticPress/pull/139/files#diff-41b7efa77b1d4d5e3fe4fbbfd4ef38b65984c74efd73ad2639f79db82e807a95L744-L746)]
* includes/classes/Indexable/Post/Post.php: [[link](https://github.com/Automattic/ElasticPress/pull/139/files#diff-7004c1b9f26ece8d68d4d3b7b822331a1f1e0cdf34a5eda1aa146d930d21ad08R361-R362)]; **please compare with the [original](https://github.com/10up/ElasticPress/blob/3.6.5/includes/classes/Indexable/Post/Post.php#L328)**
* includes/classes/Elasticsearch.php: conflicts in PHP docblocks; trivial to resolve [[link](https://github.com/Automattic/ElasticPress/pull/139/files#diff-020ebad2001f1a0d8d69510506b64d04cade5a44938dacacbbfcdb95936ca1b7L1179-R1184)]
* dist/js/comments-script.min.js: replaced with the upstream version
* dist/js/autosuggest-script.min.js: replaced with the upstream version
* CHANGELOG.md: trivial to resolve

File: includes/classes/Indexable/User/User.php [hunk 1]
```diff
@@ -728,6 +728,9 @@ public function query_db( $args ) {
 			$args['order'] = 'desc';
 		}
 
+		$orderby_args = sanitize_sql_orderby( "{$args['orderby']} {$args['order']}" );
```
Comment (pschoffer) [inline includes/classes/Indexable/User/User.php:731]: I guess this duplicates the lines below. We should drop those bellow.

**Review by pschoffer — CHANGES_REQUESTED** [review]
Looks pretty harmless :) I tested some basic stuff and it seems to be doing fine. 

I left one minor note.

File: includes/classes/Indexable/User/User.php (hunk 1 above)
Comment (sjinks) [reply includes/classes/Indexable/User/User.php:731]: I am afraid I don't understand what you mean :-(

As far as I can see, these lines come directly from the upstream:
https://github.com/10up/ElasticPress/blob/3.6.5/includes/classes/Indexable/User/User.php#L731
https://github.com/10up/ElasticPress/blob/3.6.5/includes/classes/Indexable/User/User.php#L732

Could you please clarify?

Comment (sjinks) [reply includes/classes/Indexable/User/User.php:731]: Done

**Review by rebeccahum — APPROVED** [review]

**Pull request merged**